internal/report.go

```
@@ -17,5 +31,66 @@ type Report struct {
 	// Network error.
 	Error error `json:"error,omitempty"`
 	// Extra information. Depends on the protocol.
-	Extra string `json:"extra,omitempty"`
+	Extra  string `json:"extra,omitempty"`
+	Format Format
+}
+
+func (r *Report) PrintFormatted(f Format) {
+
+	switch f {
+	case HumanFormat:
+		printHumanFormat(r)
+	case JSONFormat:
+		printJSONFormat(r)
+	case GrepFormat:
+		printGrepableFormat(r)
+	}
+
+}
+func printJSONFormat(r *Report) {
+
+	reportJSON, err := json.Marshal(r)
+	if err != nil {
+		log.Fatal(fmt.Errorf("marshaling report: %w", err))
```

Should return the error. A library shouldn't log anything.